File: Restaurant-back/src/main/java/com/spring/restaurant/entites/Product.java
package com.spring.restaurant.entites;

import jakarta.persistence.*;
import lombok.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Product extends BaseEntity
{
	@Column(length = 1000)
	private String description;

	private Double price;

	@ManyToOne
	@JoinColumn(name = "category_id", nullable = false)
	private Category category;
}

File: Restaurant-front/src/model/Product.ts
import {Category} from './Category';
import {BaseEntity} from './BaseEntity';

export class Product extends BaseEntity {
  description: string;
  price: number;
  category: Category;


  constructor(product: Partial<Product>) {
    super(product);
    if (product) {
      this.description = product.description;
      this.price = product.price;
      this.category = product.category;
    }
  }
}
